### PR TITLE
#991: DeepMHC for protein-peptide binding

### PR DESCRIPTION
**Pull request by VIGS25**
Addressing #991, this PR adds in an implementation of DeepMHC for predicting `IC50` values. 

The model needs to be run for 5000 epochs, according to the paper, but I don't have that much compute power. I could only run it for 10 epochs and the scores would be low.

But the model is running and the hyperparameters are set to what were mentioned in the paper.

### CODE REVIEW

**Comment by rbharath** [comment]
@VIGS25 This looks really cool! I'd be curious to see if the results of the original paper can be duplicated in follow-up work.

**Comment by VIGS25** [comment]
@rbharath: Yes, I'll be working on it. But the paper mentions train-valid split at every epoch of training and early stopping based on validation loss. Does TensorGraph support this right now?

**Comment by rbharath** [comment]
I think you could just write an explicit training for-loop to do this. We're going to be integrating TensorGraph and Keras much more tightly in future releases, so this is probably the right future-proofed way
